planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2676,5 +2679,86 @@ void AvoidanceModule::setDebugData(const PathShifter & shifter, const DebugData
   addAvoidPoint(debug.new_shift_points, "path_shifter_proposed_points", 0.99, 0.0, 0.0, 0.5);
 }
+double AvoidanceModule::addDistancetoReturnPoint(ObjectData & object_data) const
+{
+  using Point2d = boost::geometry::model::d2::point_xy<double>;
+  using lanelet::geometry::distance2d;
+  using lanelet::utils::getId;
+  using lanelet::utils::to2D;
+  tier4_autoware_utils::Polygon2d object_poly{};
+  util::calcObjectPolygon(object_data.object, &object_poly);
+  const auto & rh = planner_data_->route_handler;
+  auto object_pose = object_data.object.kinematics.initial_pose_with_covariance.pose;
+  object_pose.orientation = object_data.path_pose.orientation;
+  auto yaw = tf2::getYaw(object_pose.orientation) + M_PI_2;
+  Eigen::Vector3d base_unit_vec{std::cos(yaw), std::sin(yaw), 0};
+
+  double max = 0.0;
+  double min = 0.0;
+  bool flag = true;
+
+  lanelet::ConstLanelets checklanelets;
+  lanelet::ConstLanelet nextlanelet;
+  lanelet::ConstLanelets pretlanelets;
+
+  checklanelets.push_back(object_data.overhang_lanelet);
+  if (rh->getNextLaneletWithinRoute(object_data.overhang_lanelet, &nextlanelet)) {
+    checklanelets.push_back(nextlanelet);
+  }
+
+  if (rh->getPreviousLaneletsWithinRoute(object_data.overhang_lanelet, &pretlanelets)) {
+    checklanelets.push_back(pretlanelets.front());
+  }
+
+  std::vector<Point2d> collision_points;
+  std::vector<Point2d> in_lane_points;
+  std::vector<Point2d> check_points;
+  for (auto lanelet : checklanelets) {
+    std::vector<Point2d> right_collision_points;
+    std::vector<Point2d> left_collision_points;
+    boost::geometry::intersection(
+      object_poly, to2D(lanelet.rightBound().basicLineString()), right_collision_points);
+    boost::geometry::intersection(
+      object_poly, to2D(lanelet.leftBound().basicLineString()), left_collision_points);
+    collision_points.insert(
+      collision_points.end(), right_collision_points.begin(), right_collision_points.end());
+    collision_points.insert(
+      collision_points.end(), left_collision_points.begin(), left_collision_points.end());
+  }
+
+  geometry_msgs::msg::Pose target_pose;
+  for (const auto & p : object_poly.outer()) {
+    const auto target_point = tier4_autoware_utils::createPoint(p.x(), p.y(), 0.0);
+
+    target_pose.position = target_point;
+    for (auto lanelet : checklanelets) {
+      if (lanelet::utils::isInLanelet(target_pose, lanelet, 0.001)) {
+        Point2d point(p.x(), p.y());
+        in_lane_points.push_back(point);
+      }
+    }
+  }
+
+  check_points.insert(check_points.end(), collision_points.begin(), collision_points.end());
+  check_points.insert(check_points.end(), in_lane_points.begin(), in_lane_points.end());
 
+  for (auto & check_point : check_points) {
+    const auto dx = check_point.x() - object_pose.position.x;
+    const auto dy = check_point.y() - object_pose.position.y;
+    const Eigen::Vector3d diff_vec{dx, dy, 0};
+    const Eigen::Vector3d cross_vec = base_unit_vec.cross(diff_vec);
```

Comment:
@shulanbushangshu 
Thank you for description can you also add how long is "distance_add_to_returnpoint" ?

and I suggest these changes (please give me opinion if I'm wrong.)
- I think this name( "distance_add_to_returnpoint") should be something like longitudinal_collision_margin
- longitudinal margin should be added to NOT closer footprint point because it's a bug which should be fixed.
![image](https://user-images.githubusercontent.com/65527974/171116694-a6fb825b-fc84-4c71-939a-a5be7f21cfea.png)

Reply:
@taikitanaka3
Yes,you are right.In the changes,the "distance_add_to_returnpoint" is calculated according to the check points (including the intersections of the obstacle and the lane boundary, and the contour point of the obstacle within the lane).
The "distance_add_to_returnpoint" can be as longitudinal_collision_margin.
The schematic of the  "distance_add_to_returnpoint" is shown below：
![distance](https://user-images.githubusercontent.com/102840938/171134680-95943418-72cc-4e34-9374-bf2d530cdf79.png)

In the code,I use the check points to calculate the maximum distance in the "object_closest_pose" point direction.The maximum distance is the "distance_add_to_returnpoint"

Reply:
@shulanbushangshu 
can you add better name for check point or add description to readme?

@zulfaqar-azmi-t4 
can you share figures with @shulanbushangshu used in readme?

Reply:
@shulanbushangshu 
can you add better name for check point or add description to readme?

@zulfaqar-azmi-t4 
can you share figures with @shulanbushangshu used in readme?

Reply:
@taikitanaka3 
Yes, this is the drawio file.

[feature_behavior_path_planner.zip](https://github.com/autowarefoundation/autoware.universe/files/8821650/feature_behavior_path_planner.zip)